page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/utils/Wait.java

```diff
@@ -0,0 +1,634 @@
+package ru.sbtqa.tag.pagefactory.utils;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.aeonbits.owner.ConfigFactory;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedCondition;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import ru.sbtqa.tag.pagefactory.environment.Environment;
+import ru.sbtqa.tag.pagefactory.properties.Configuration;
+
+public class Wait {
+
+    private static final Configuration PROPERTIES = ConfigFactory.create(Configuration.class);
+
+    /**
+     * Wait until the element becomes visible
+     *
+     * @param xpath element xpath
+     */
+    public static void visibility(String xpath) {
+        visibility(xpath, "Element with xpath \"" + xpath + "\" is not visible", PROPERTIES.getTimeout());
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Define a constant instead of duplicating this literal "\" is not visible" 3 times. ([squid:S1192](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1192))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=111&issues=AWdUlBbny5Us1olWVmP2&open=AWdUlBbny5Us1olWVmP2)

[ISSUE_KEY]: AWdUlBbny5Us1olWVmP2